packages/pigeon/platform_tests/flutter_null_safe_unit_tests/lib/null_safe_pigeon.dart

```diff
@@ -28,98 +28,64 @@ class SearchReply {
-    return Nested()
-      ..request = pigeonMap['request'] != null
-          ? SearchRequest.decode(pigeonMap['request']!)
-          : null;
+    return SearchReplies()..replies = pigeonMap['replies'] as List<Object?>?;
   }
 }
 
-abstract class FlutterSearchApi {
-  SearchReply search(SearchRequest arg);
-  static void setup(FlutterSearchApi? api) {
-    {
-      const BasicMessageChannel<Object?> channel = BasicMessageChannel<Object?>(
-          'dev.flutter.pigeon.FlutterSearchApi.search', StandardMessageCodec());
-      if (api == null) {
-        channel.setMessageHandler(null);
-      } else {
-        channel.setMessageHandler((Object? message) async {
-          assert(message != null,
-              'Argument for dev.flutter.pigeon.FlutterSearchApi.search was null. Expected SearchRequest.');
-          final SearchRequest input = SearchRequest.decode(message!);
-          final SearchReply output = api.search(input);
-          return output.encode();
-        });
-      }
-    }
+class SearchRequests {
+  List<Object?>? requests;
+
+  Object encode() {
+    final Map<Object?, Object?> pigeonMap = <Object?, Object?>{};
+    pigeonMap['requests'] = requests;
+    return pigeonMap;
   }
-}
 
-class NestedApi {
-  Future<SearchReply> search(Nested arg) async {
-    final Object encoded = arg.encode();
-    const BasicMessageChannel<Object?> channel = BasicMessageChannel<Object?>(
-        'dev.flutter.pigeon.NestedApi.search', StandardMessageCodec());
-    final Map<Object?, Object?>? replyMap =
-        await channel.send(encoded) as Map<Object?, Object?>?;
-    if (replyMap == null) {
-      throw PlatformException(
-        code: 'channel-error',
-        message: 'Unable to establish connection on channel.',
-        details: null,
-      );
-    } else if (replyMap['error'] != null) {
-      final Map<Object?, Object?> error =
-          replyMap['error']! as Map<Object?, Object?>;
-      throw PlatformException(
-        code: error['code']! as String,
-        message: error['message'] as String?,
-        details: error['details'],
-      );
-    } else {
-      return SearchReply.decode(replyMap['result']!);
-    }
+  static SearchRequests decode(Object message) {
+    final Map<Object?, Object?> pigeonMap = message as Map<Object?, Object?>;
+    return SearchRequests()..requests = pigeonMap['requests'] as List<Object?>?;
   }
 }
 
 class Api {
-  Future<void> initialize() async {
-    const BasicMessageChannel<Object?> channel = BasicMessageChannel<Object?>(
-        'dev.flutter.pigeon.Api.initialize', StandardMessageCodec());
+  /// Constructor for [Api].  The [binaryMessenger] named argument is
+  /// available for dependency injection.  If it is left null, the default
+  /// BinaryMessenger will be used which routes to the host platform.
+  Api({BinaryMessenger? binaryMessenger}) : _binaryMessenger = binaryMessenger;
+
+  final BinaryMessenger? _binaryMessenger;
```

Comment:
where does the defaulting happen? I'd be generally expecting this to be non-null.

Reply:
`null` has a special interpretation in the channels code to use the default binary messenger.  That's what we were providing previously by not specifying the messenger.

Reply:
ah ok